core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/atom/textfield/TextFieldCommon.kt

```diff
@@ -5,6 +5,24 @@ import androidx.compose.runtime.Composable
 import androidx.compose.ui.tooling.preview.Preview
 import app.k9mail.core.ui.compose.theme.PreviewWithThemes
 
+private val LINE_BREAK = "[\\r\\n]".toRegex()
+
+internal fun fixSingleLineBehavior(isSingleLine: Boolean, onValueChange: (String) -> Unit): (String) -> Unit {
```

Comment:
This could be simplified to:

```suggestion
internal fun fixSingleLineBehavior(isSingleLine: Boolean, onValueChange: (String) -> Unit): (String) -> Unit {
    return { value ->
        if (isSingleLine) {
            onValueChange(value.replace(LINE_BREAK, replacement = ""))
        } else {
            onValueChange(value)
        }
    }
}
```

Reply:
I don't like how `isSingleLine` is checked inside the lambda instead of outside. Given that `fixSingleLineBehavior()` was only used once, I decided to move the logic to the call site.

https://github.com/thunderbird/thunderbird-android/blob/9bef2a4b0651f7e219a3ebd94e40b6ac79ddd9de/core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/atom/textfield/TextFieldOutlined.kt#L29

Reply:
Fine with me